src/app/archive/zkapp_tables.sql

```diff
@@ -95,21 +95,30 @@ CREATE TABLE zkapp_verification_keys
 
 CREATE TYPE zkapp_auth_required_type AS ENUM ('none', 'either', 'proof', 'signature', 'both', 'impossible');
 
+CREATE TABLE protocol_versions
+( id               serial PRIMARY KEY
+, transaction      int    NOT NULL
+, network          int    NOT NULL
+, patch            int    NOT NULL
+, UNIQUE (transaction,network,patch)
+);
+
 CREATE TABLE zkapp_permissions
-( id                       serial                PRIMARY KEY
-, edit_state               zkapp_auth_required_type    NOT NULL
-, send                     zkapp_auth_required_type    NOT NULL
-, receive                  zkapp_auth_required_type    NOT NULL
-, access                   zkapp_auth_required_type    NOT NULL
-, set_delegate             zkapp_auth_required_type    NOT NULL
-, set_permissions          zkapp_auth_required_type    NOT NULL
-, set_verification_key     zkapp_auth_required_type    NOT NULL
-, set_zkapp_uri            zkapp_auth_required_type    NOT NULL
-, edit_action_state      zkapp_auth_required_type    NOT NULL
-, set_token_symbol         zkapp_auth_required_type    NOT NULL
-, increment_nonce          zkapp_auth_required_type    NOT NULL
-, set_voting_for           zkapp_auth_required_type    NOT NULL
-, set_timing               zkapp_auth_required_type    NOT NULL
+( id                           serial                      PRIMARY KEY
+, edit_state                   zkapp_auth_required_type    NOT NULL
+, send                         zkapp_auth_required_type    NOT NULL
+, receive                      zkapp_auth_required_type    NOT NULL
+, access                       zkapp_auth_required_type    NOT NULL
+, set_delegate                 zkapp_auth_required_type    NOT NULL
+, set_permissions              zkapp_auth_required_type    NOT NULL
+, set_verification_key         zkapp_auth_required_type    NOT NULL
+, set_verification_key_version int                         NOT NULL REFERENCES protocol_versions(id)
```

Comment:
should this be just the transaction version?

if it is a reference to the entire protocol version, the column name should end in `_id`

Reply:
Yeah, since the comparison just need `transaction`, it seems like we don't need the full protocol_version. The same applies to the ocaml `Permission.t` type. Let me ask nathan whether we want to keep the entire protocol version or we just need `transaction` here.

Reply:
Have you gotten resolution on this issue?